### PR TITLE
hail updates

### PR DESCRIPTION
**Pull request by RoriCremer**

### CODE REVIEW

**Review by mcovarr — APPROVED** [review]
LGTM

File: scripts/variantstore/wdl/extract/hail_create_vat_inputs.py [hunk 1]
```diff
@@ -124,37 +124,52 @@ def write_sites_only_vcf(ac_an_af_split, sites_only_vcf_path):
...
-    ]
-    transformed_vds=vds
-    for transform in transforms:
-        transformed_vds = transform(transformed_vds)
+    n_parts = vds.variant_data.n_partitions()
+    n_rounds = 5
+    parts_per_round = n_parts // n_rounds
+    for i in range(n_rounds):
+        part_range = range(i*parts_per_round, min((i+1)*parts_per_round, n_parts))
+        vds = hl.vds.VariantDataset(
```
Comment (mcovarr) [inline scripts/variantstore/wdl/extract/hail_create_vat_inputs.py:143]: this is clobbering the original value of `vds`, should probably be a new variable like `vds_range` or something

File: scripts/variantstore/wdl/extract/hail_create_vat_inputs.py [hunk 2]
```diff
@@ -124,37 +124,52 @@ def write_sites_only_vcf(ac_an_af_split, sites_only_vcf_path):
 
     ht = ac_an_af_rows.rows()
     ht = ht.filter(ht.alleles[1] != "*") # remove spanning deletions
+    ht = ht.checkpoint("gs://fc-eada2674-7c2b-42a6-8db3-0246872596dc/checkpoint")
```
Comment (mcovarr) [inline scripts/variantstore/wdl/extract/hail_create_vat_inputs.py:127]: not sure we still need this (or that we don't)

Comment (RoriCremer) [reply scripts/variantstore/wdl/extract/hail_create_vat_inputs.py:127]: yeah, I will leave that question for Patrick!

File: scripts/variantstore/wdl/extract/hail_create_vat_inputs.py [hunk 3]
```diff
@@ -124,37 +124,52 @@ def write_sites_only_vcf(ac_an_af_split, sites_only_vcf_path):
...
 
-    # for debugging information
-    add_variant_tracking_info(mt, sites_only_vcf_path)
+        # for debugging information -- remove for now to get us through Echo
+        # add_variant_tracking_info(mt, sites_only_vcf_path)
 
-    # create a sites only VCF (that is hard filtered!) and that can be made into a custom annotations TSV for Nirvana to use with AC, AN, AF, SC for all subpopulations and populations
-    write_sites_only_vcf(mt, sites_only_vcf_path)
+        # create a sites only VCF (that is hard filtered!) and that can be made into a custom annotations TSV for Nirvana to use with AC, AN, AF, SC for all subpopulations and populations
+        write_sites_only_vcf(mt, sites_only_vcf_path)
```
Comment (RoriCremer) [inline scripts/variantstore/wdl/extract/hail_create_vat_inputs.py:172]: I was actually more worried about the vcf writing getting clobbered. but maybe I dont understand how this works

File: scripts/variantstore/wdl/extract/hail_create_vat_inputs.py [hunk 4]
```diff
@@ -124,37 +124,52 @@ def write_sites_only_vcf(ac_an_af_split, sites_only_vcf_path):
 
     ht = ac_an_af_rows.rows()
     ht = ht.filter(ht.alleles[1] != "*") # remove spanning deletions
+    ht = ht.checkpoint("gs://fc-eada2674-7c2b-42a6-8db3-0246872596dc/checkpoint")
     # create a filtered sites only VCF
-    hl.export_vcf(ht, sites_only_vcf_path)
+    hl.export_vcf(ht, sites_only_vcf_path, parallel='separate_header')
 
 def add_variant_tracking_info(mt, sites_only_vcf_path):
```
Comment (mcovarr) [inline scripts/variantstore/wdl/extract/hail_create_vat_inputs.py:131]: this one doesn't seem to be called anymore

File: scripts/variantstore/wdl/extract/hail_create_vat_inputs.py (hunk 3 above)
Comment (mcovarr) [reply scripts/variantstore/wdl/extract/hail_create_vat_inputs.py:172]: uh yeah good point, the destination path might also need to become a function of the range index...

File: scripts/variantstore/wdl/extract/hail_create_vat_inputs.py (hunk 4 above)
Comment (gbggrant) [reply scripts/variantstore/wdl/extract/hail_create_vat_inputs.py:129]: Do you want to remove this method? (seems to be referenced in commented out debugging code though?)

**Review by gbggrant — APPROVED** [review]
Looks good, but I'm not a hail guru.

File: scripts/variantstore/wdl/extract/hail_create_vat_inputs.py [hunk 5]
```diff
@@ -110,7 +110,7 @@ def write_sites_only_vcf(ac_an_af_split, sites_only_vcf_path):
         pop_info_fields[f'Hom_{pop}'] = ac_an_af_split.call_stats_by_pop.get(pop).homozygote_count[ac_an_af_split.row.a_index]
 
 
-    ac_an_af_rows = ac_an_af_split.annotate_rows(
+    ac_an_af_rows = ac_an_af_split.annotate(
```
Comment (mcovarr) [inline scripts/variantstore/wdl/extract/hail_create_vat_inputs.py:113]: Is this still a "rows" object? If not, its name probably shouldn't say "rows".

File: scripts/variantstore/wdl/extract/hail_create_vat_inputs.py [hunk 6]
```diff
@@ -122,39 +122,55 @@ def write_sites_only_vcf(ac_an_af_split, sites_only_vcf_path):
...
-    # for debugging information
-    add_variant_tracking_info(mt, sites_only_vcf_path)
+        # for debugging information -- remove for now to get us through Echo
+        # add_variant_tracking_info(mt, sites_only_vcf_path)
 
     # create a sites only VCF (that is hard filtered!) and that can be made into a custom annotations TSV for Nirvana to use with AC, AN, AF, SC for all subpopulations and populations
-    write_sites_only_vcf(mt, sites_only_vcf_path)
+    ht_list = [hl.read_table(ht_path) for ht_path in ht_paths]
+    ht_all = ht_list[0].union(*ht_list[1:])
+    write_sites_only_vcf(ht_all, sites_only_vcf_path)
```
Comment (mcovarr) [inline scripts/variantstore/wdl/extract/hail_create_vat_inputs.py:173]: this is magical to me... can we add some comments to explain what's happening here?

**Pull request merged**